processor/src/decoder/mod.rs

```diff
@@ -586,11 +591,16 @@ impl Decoder {
     /// Adds an operation to the operations vector in debug mode.
     #[inline(always)]
     fn append_operation(&mut self, op: Operation) {
-        if self.in_debug_mode {
-            self.operations.push(op);
+        if self.debug_info.in_debug_mode() {
+            self.debug_info.append_operation(op);
         }
     }
```

Comment:
Let's remove this method from here. It is a private method and I think we can just as easily call `self.debug_info.append_operation()` where it is being called.

Reply:
Thanks. Done.